commit 171ff8faf6a902e71aa9b153397e4e15540625e9 (refs/changes/48/3048/5)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2018-06-12T00:02:58+00:00 (1 year, 4 months ago)
    
    OS-6497 Non-primary NICs of "joyent" branded zones don't work with just "addrconf"
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index 5ee8e9a1..32a0cafe 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -3429,6 +3429,10 @@ function writeZoneNetfiles(payload, log, callback)
                         static_ips);
                 }
 
+                if (addrconf_found) {
+                    replaceFile(zoneroot, '/etc/addrconf.' + n.interface, '');
+                }
+
                 if (n.hasOwnProperty('primary') && !primary_found) {
                     // Only allow one primary network
                     primary_found = true;
@@ -3439,10 +3443,6 @@ function writeZoneNetfiles(payload, log, callback)
                     if (dhcp_found) {
                         replaceFile(zoneroot, '/etc/dhcp.' + n.interface, '');
                     }
-                    if (addrconf_found) {
-                        replaceFile(zoneroot, '/etc/addrconf.' + n.interface,
-                            '');
-                    }
                 }
             });
         }
